Sources/TableViewDriver.swift

```diff
@@ -182,40 +189,56 @@ open class TableViewDriver: NSObject {
 
         guard let newModel = newModel else { return }
 
-        if self._automaticDiffingEnabled {
+        if self._automaticDiffingEnabled, self._lightweightDiffing {
+            let old = oldModel?.sectionModels.reduce(into: 0) { $0 += $1.cellViewModels.count }
+            let new = newModel.sectionModels.reduce(into: 0) { $0 += $1.cellViewModels.count }
 
-            let visibleIndexPaths = tableView.indexPathsForVisibleRows ?? []
-            let old: [DiffableTableSectionViewModel] = oldModel?.sectionModelsForDiffing(inVisibleIndexPaths: visibleIndexPaths) ?? []
-            let changeset = StagedChangeset(
-                source: old,
-                target: newModel.sectionModelsForDiffing(inVisibleIndexPaths: visibleIndexPaths)
-            )
-            if changeset.isEmpty {
-                self._tableViewModel = newModel
+            self._tableViewModel = newModel
+            if old == new {
+                self.refreshViews(refreshContext: .contentOnly)
             } else {
```

Comment:
New behavior is opt in for folks that set `automaticDiffing` to true and also set the new parameter `lightweightDiffing` to true.